inkscape.1.in
.\" Automatically generated by Pod::Man v1.37, Pod::Parser v1.13
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  | will give a
.\" real vertical bar.  \*(C+ will give a nicer C++.  Capital omega is used to
.\" do unbreakable dashes and therefore won't be available.  \*(C` and \*(C'
.\" expand to `' in nroff, nothing in troff, for use with C<>.
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "INKSCAPE 1"
.TH INKSCAPE 1 "2004-09-08" "perl v5.8.1" "0.40"
.SH "NAME"
Inkscape \- an SVG (Scalable Vector Graphics) vector editor.
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\f(CW\*(C`inkscape [options] [filename ...]\*(C'\fR
.PP
options:
.PP
.Vb 2
\&    \-?, \-\-help          
\&        \-\-usage
.Ve
.PP
.Vb 2
\&    \-f, \-\-file=FILENAME               
\&    \-s, \-\-slideshow
.Ve
.PP
.Vb 9
\&    \-e, \-\-export\-png=FILENAME         
\&    \-a, \-\-export\-area=x0:y0:x1:y1     
\&    \-i, \-\-export\-id=ID     
\&    \-t, \-\-export\-use\-hints
\&    \-b, \-\-export\-background=COLOR     
\&    \-y, \-\-export\-background\-opacity=VALUE     
\&    \-d, \-\-export\-dpi=DPI              
\&    \-w, \-\-export\-width=WIDTH          
\&    \-h, \-\-export\-height=HEIGHT
.Ve
.PP
.Vb 1
\&    \-l, \-\-export\-plain\-svg=FILENAME
.Ve
.PP
.Vb 1
\&    \-p, \-\-print=PRINTER
.Ve
.PP
.Vb 1
\&    \-x, \-\-bitmap\-icons
.Ve
.PP
.Vb 2
\&    \-g, \-\-with\-gui                    
\&    \-z, \-\-without\-gui
.Ve
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBInkscape\fR is a \s-1GUI\s0 editor for \fBScalable Vector Graphics (\s-1SVG\s0)\fR format
drawing files, with capabilities similar to \fBAdobe Illustrator\fR,
\&\fBCorelDraw\fR, \fBVisio\fR, etc. Inkscape features include versatile
shapes, bezier paths, freehand drawing, multiline text, 
alpha blending, transforms, gradients, node
editing, SVG-to-PNG export, grouping, patterns, and more.  The interface is
designed to be comfortable and efficient for skilled users, while
remaining conformant to \fB\s-1GNOME\s0\fR standards so that users familiar with
other \s-1GNOME\s0 applications can learn its interface rapidly.
.PP
\&\fB\s-1SVG\s0\fR is a W3C standard \s-1XML\s0 format for 2D vector drawing. It allows
defining objects in the drawing using points, paths, and primitive
shapes.  Colors, fonts, stroke width, and so forth are specified as
`style' attributes to these objects.  The intent is that since \s-1SVG\s0 is a
standard, and since its files are text/xml instead of binary, it will be
possible to use \s-1SVG\s0 files in a sizeable number of programs and for a
wide range of uses.
.PP
\&\fBInkscape\fR uses \s-1SVG\s0 as its native document format, and has the goal of
becoming the most fully compliant drawing program for \s-1SVG\s0 files
available in the Open Source community.
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "\fB\-?\fR, \fB\-\-help\fR" 8
.IX Item "-?, --help"
Show help message
.IP "\fB\-a\fR \fIx0:y0:x1:y1\fR, \fB\-\-export\-area\fR=\fIx0:y0:x1:y1\fR" 8
.IX Item "-a x0:y0:x1:y1, --export-area=x0:y0:x1:y1"
Exported area in \s-1SVG\s0 pixels (anonymous length units normally used in Inkscape \s-1SVG\s0). The default is to export the entire document page. The 0,0 is
the lower-left corner.
.IP "\fB\-b\fR \fI\s-1COLOR\s0\fR, \fB\-\-export\-background\fR=\fI\s-1COLOR\s0\fR" 8
.IX Item "-b COLOR, --export-background=COLOR"
Background color of exported \s-1PNG\s0. This may be any \s-1SVG\s0 supported color string, for example \*(L"#ff007f\*(R" or \*(L"rgb(255, 0, 128)\*(R". If not set, the page color set in Inkscape in the Document Options dialog will be used (stored in the pagecolor= attribute of sodipodi:namedview).
.IP "\fB\-d\fR \fI\s-1DPI\s0\fR, \fB\-\-export\-dpi\fR=\fI\s-1DPI\s0\fR" 8
.IX Item "-d DPI, --export-dpi=DPI"
The resolution used for bitmap export. The default is 72 which corresponds to 1 \s-1SVG\s0 pixel exporting to 1 bitmap pixel. This value overrides the \s-1DPI\s0 hint if used with \-\-export\-use\-hints.
.IP "\fB\-e\fR \fI\s-1FILENAME\s0\fR, \fB\-\-export\-png\fR=\fI\s-1FILENAME\s0\fR" 8
.IX Item "-e FILENAME, --export-png=FILENAME"
Specify the filename for \s-1PNG\s0 export. If exists, the file will be overwritten without asking.
.IP "\fB\-f\fR \fI\s-1FILENAME\s0\fR, \fB\-\-file\fR=\fI\s-1FILENAME\s0\fR" 8
.IX Item "-f FILENAME, --file=FILENAME"
Open specified document(s). Option string may be omitted, i.e. you can list the filenames without \-f.
.IP "\fB\-g\fR, \fB\-\-with\-gui\fR" 8
.IX Item "-g, --with-gui"
Try to use the \s-1GUI\s0 (on Unix, use the X server even if \f(CW$DISPLAY\fR is not set).
.IP "\fB\-h\fR \fI\s-1HEIGHT\s0\fR, \fB\-\-export\-height\fR=\fI\s-1HEIGHT\s0\fR" 8
.IX Item "-h HEIGHT, --export-height=HEIGHT"
The height of generated bitmap in pixels. This value overrides the \-\-export\-dpi setting (or the \s-1DPI\s0 hint if used with \-\-export\-use\-hints). 
.IP "\fB\-i\fR \fI\s-1ID\s0\fR, \fB\-\-export\-id\fR=\fI\s-1ID\s0\fR" 8
.IX Item "-i ID, --export-id=ID"
The id attribute value of the object that you want to export from the document. Exported is the bounding box of the object, so with this option, \-\-export\-area is ignored. 
.IP "\fB\-l\fR, \fB\-\-export\-plain\-svg\fR=\fI\s-1FILENAME\s0\fR" 8
.IX Item "-l, --export-plain-svg=FILENAME"
Export document(s) to plain \s-1SVG\s0 format, without sodipodi: or inkscape: namespaces and without \s-1RDF\s0 metadata.
.IP "\fB\-p\fR \fI\s-1PRINTER\s0\fR, \fB\-\-print\fR=\fI\s-1PRINTER\s0\fR" 8
.IX Item "-p PRINTER, --print=PRINTER"
Print document(s) to the specified printer using `lpr \-P \s-1PRINTER\s0'.
Alternatively, use `| \s-1COMMAND\s0' to specify a different command to pipe to,
or use `> \s-1FILENAME\s0' to write the PostScript output to a file instead of printing.
Remember to do appropriate quoting for your shell, e.g.
.Sp
inkscape \-\-print='| ps2pdf \- mydoc.pdf' mydoc.svg
.IP "\fB\-s\fR, \fB\-\-slideshow\fR" 8
.IX Item "-s, --slideshow"
Show given files one by one, switching to the next one by any key or mouse event.
.IP "\fB\-t\fR, \fB\-\-export\-use\-hints\fR" 8
.IX Item "-t, --export-use-hints"
Use export filename and \s-1DPI\s0 hints stored in the exported object (only with \-\-export\-id). These hints are set automatically when you export selection from within Inkscape. So, for example, if you export a shape with id=\*(L"path231\*(R" as /home/me/shape.png at 300 dpi from document.svg using Inkscape \s-1GUI\s0 and then save the document, later you will be able to reexport that shape to the same file with the same resolution simply with 
.Sp
inkscape \-i path231 \-t document.svg 
.Sp
If you use \-\-export\-dpi, \-\-export\-width, or \-\-export\-height with this option, the \s-1DPI\s0 hint will be ignored and the value from the command line will be used. If you use \-\-export\-png with this option, the filename hint will be ignored and the filename from the command line will be used. 
.IP "\fB\-w\fR \fI\s-1WIDTH\s0\fR, \fB\-\-export\-width\fR=\fI\s-1WIDTH\s0\fR" 8
.IX Item "-w WIDTH, --export-width=WIDTH"
The width of generated bitmap in pixels. This value overrides the \-\-export\-dpi setting (or the \s-1DPI\s0 hint if used with \-\-export\-use\-hints). 
.IP "\fB\-x\fR, \fB\-\-bitmap\-icons\fR" 8
.IX Item "-x, --bitmap-icons"
Use bitmap (\s-1XPM\s0) icons instead of \s-1SVG\s0 ones. By default, Inkscape raterizes icons on-the-fly 
from share/inkscape/icons/icons.svg. Use this option to turn this off if there is any problem with \s-1SVG\s0 icons. 
.IP "\fB\-y\fR \fI\s-1VALUE\s0\fR, \fB\-\-export\-background\-opacity\fR=\fI\s-1VALUE\s0\fR" 8
.IX Item "-y VALUE, --export-background-opacity=VALUE"
Opacity of the background of exported \s-1PNG\s0. This may be a value either between 0.0 and 1.0 (0.0 meaning full transparency, 1.0 full opacity) or between 1 and 255 (255 meaning full opacity). If not set and the \-b option is not used, the page opacity set in Inkscape in the Document Options dialog will be used (stored in the inkscape:pageopacity= attribute of sodipodi:namedview). If not set but the \-b option is used, the value of 255 (full opacity) will be used. 
.IP "\fB\-z\fR, \fB\-\-without\-gui\fR" 8
.IX Item "-z, --without-gui"
Do not open the \s-1GUI\s0 (on Unix, do not use X server); only process the files from console. This is assumed for \-p, \-e, and \-l options. 
.IP "\fB\-\-usage\fR" 8
.IX Item "--usage"
Display a brief usage message.
.SH "KEYBINDINGS"
.IX Header "KEYBINDINGS"
To get a complete list of keyboard and mouse shortcuts, view doc/keys.html, or use the Keys and Mouse command in Help menu from the \s-1GUI\s0 to see an \s-1SVG\s0 chart.
.SH "CONFIGURATION"
.IX Header "CONFIGURATION"
The preferences.xml configuration file located in ~/.inkscape/ is used
to customize the application settings for the user.  
.IP "\fBinterface\fR" 8
.IX Item "interface"
The \fBinterface\fR element(s) of the config file is used to set parameters
related to the \s-1GUI\s0 interface, such as the open/closed status of various
\&\s-1GUI\s0 elements, etc. 
.ie n .IP "\fBgroup id=""documents""\fR" 8
.el .IP "\fBgroup id=``documents''\fR" 8
.IX Item "group id=documents"
The documents group is used for containing the recent files list.  Each
document is listed with its uri (path) and name indicated.
.ie n .IP "\fBgroup id=""template""\fR" 8
.el .IP "\fBgroup id=``template''\fR" 8
.IX Item "group id=template"
The template group is used for storing parameters related to blank
documents. 
.ie n .IP "\fBgroup id=""tools""\fR" 8
.el .IP "\fBgroup id=``tools''\fR" 8
.IX Item "group id=tools"
The tools group is used for storing the user style preferences for
different event contexts (i.e., shapes, freehand or calligraphic stroke
properties, etc.).
.ie n .IP "\fBgroup id=""palette""\fR" 8
.el .IP "\fBgroup id=``palette''\fR" 8
.IX Item "group id=palette"
The palette group allows setting of dash styles.  This allows you to
define the stroke lengths for different kinds of dashes.
.ie n .IP "\fBgroup id=""dialogs""\fR" 8
.el .IP "\fBgroup id=``dialogs''\fR" 8
.IX Item "group id=dialogs"
The dialogs group allows persisting the position and width of each of
the dialogs in the application, so that they'll start up in the last
place the user had them at the next time the app is run.
.ie n .IP "\fBgroup id=""printing""\fR" 8
.el .IP "\fBgroup id=``printing''\fR" 8
.IX Item "group id=printing"
The printing group is for storing different printer settings.  Each
setting is identified with an id.  Properties include \fIbitmap\fR
(true/false), \fIresolution\fR, and \fIdestination\fR.
.ie n .IP "\fBgroup id=""options""\fR" 8
.el .IP "\fBgroup id=``options''\fR" 8
.IX Item "group id=options"
The options group allows persisting various user selected options
including \fInudgeddistance\fR, \fIrotationstep\fR, \fIcursortolerance\fR, and
\&\fIdragtolerance\fR. 
.SH "DIAGNOSTICS"
.IX Header "DIAGNOSTICS"
The program returns zero on success or non-zero on failure.
.PP
A variety of error messages and warnings are printed to \s-1STDERR\s0 or
\&\s-1STDOUT\s0.  If the program behaves erratically with a particular \s-1SVG\s0 file
or crashes, it is sometimes useful to look at this output for clues.
For example, it may indicate if a given \s-1SVG\s0 file contains elements or
attributes that are not yet supported by the codebase.
.SH "EXAMPLES"
.IX Header "EXAMPLES"
While obviously \fBInkscape\fR is primarily intended as a \s-1GUI\s0 application,
it can be used for doing \s-1SVG\s0 processing on the commandline as well.
.PP
Print an \s-1SVG\s0 file:
.PP
.Vb 1
\&    inkscape filename.svg \-p \[aq]| lpr\[aq]
.Ve
.PP
Export an \s-1SVG\s0 file into \s-1PNG:\s0
.PP
.Vb 1
\&    inkscape filename.svg \-\-export\-png=filename.png \-w600 \-h400
.Ve
.PP
Convert an Inkscape \s-1SVG\s0 document to plain \s-1SVG:\s0  
.PP
.Vb 1
\&    inkscape filename1.svg \-\-export\-plain\-svg=filename2.svg
.Ve
.SH "ENVIRONMENT"
.IX Header "ENVIRONMENT"
\&\fB\s-1DISPLAY\s0\fR to get the default host and display number.
.PP
\&\fB\s-1TMPDIR\s0\fR to set the default path of the directory to use for temporary
files.  The directory must exist.
.SH "FILES"
.IX Header "FILES"
\&\fB$HOME\fR/.inkscape/preferences.xml \- The user's preference settings. 
.PP
\&\fB$HOME\fR/.inkscape/extensions.xml \- The filter programs to be used in
the application.
.SH "OTHER INFO"
.IX Header "OTHER INFO"
The canonical place to find \fBInkscape\fR info is at
http://www.inkscape.org/.  The website includes links to other relevant
documentation, tutorials, user manual, examples, mailing list archives,
the latest released version of the program, and more. 
.SH "SEE ALSO"
.IX Header "SEE ALSO"
\&\fIgimp\fR\|(1), autotrace, potrace, frontline, ill2svg, \fIrsvg\fR\|(1), \fIxfig\fR\|(1), sodipodi,
karbon14, dia(1X), batik.
.PP
\&\s-1SVG\s0 compliance test suite:  http://www.w3.org/Graphics/SVG/Test/
.PP
\&\s-1SVG\s0 validator:  http://jiggles.w3.org/svgvalidator/
.PP
\&\fIScalable Vector Graphics (\s-1SVG\s0) 1.1 Specification\fR
\&\fIW3C Recommendation 14 January 2003\fR
<http://www.w3.org/TR/SVG11/>
.PP
\&\fIScalable Vector Graphics (\s-1SVG\s0) 1.2 Specification\fR
\&\fIW3C Working Draft 13 November 2003\fR
<http://www.w3.org/TR/SVG12/>
.PP
\&\fI\s-1SVG\s0 1.1/1.2/2.0 Requirements\fR
\&\fIW3C Working Draft 22 April 2002\fR
<http://www.w3.org/TR/SVG2Reqs/>
.PP
\&\fIDocument Object Model (\s-1DOM\s0): Level 2 Core\fR
\&\fIArnaud Le Hors et al editors, W3C\fR
<http://www.w3.org/TR/DOM\-Level\-2\-Core/>
.SH "NOTES"
.IX Header "NOTES"
Ctrl, Shift, Alt modify dragging and clicking behaviour in most tools. Watch the statusbar
for helpful hints.
.PP
To rotate an object, click it to select with Selector tool, then click again to enable
rotation handles.
.PP
The Import command works with most bitmap formats (\s-1PNG\s0, \s-1BMP\s0, \s-1JPG\s0, \s-1XPM\s0, \s-1GIF\s0 etc.) and \s-1SVG\s0
vector images. The Open command can handle only \s-1SVG\s0 files.
.PP
The Export command only exports bitmaps in \s-1PNG\s0 format. To export into \s-1PS\s0, \s-1EPS\s0 and other
vector formats, use Save As.
.PP
A Wacom tablet will work with Inkscape if it is configured to act as a regular pointing
device. We do not support pressure/tilt sensitivity or any tablet-specific
features yet.
.PP
It works \*(L"out of the box\*(R" with Apple's X server, but with XFree86 you
will need to have the \*(L"SendCoreEvents\*(R" option turned on if you do not
already.  It does not seem to work as well under Windows.
.SH "BUGS"
.IX Header "BUGS"
Many bugs are known; please refer to the website for reviewing the
reported ones and to report newly found issues.  Here are some workarounds:
.PP
When importing svg generated with ill2svg, all paths are usually grouped.
Ungroup these manually.
.PP
If you use gradients and stretch the icon of the saved file in Nautilus,
you'll notice that the gradients don't render right.  This is a
well-known problem in Nautilus/RSVG and can be worked around by
specifying the gradient as 'userspaceinuse' in the gradient editor.
.PP
See also the Known Issues section in the Release Notes for your version (file `\s-1NEWS\s0').
.SH "AUTHORS"
.IX Header "AUTHORS"
This codebase owes its existance to a large number of contributors
throughout its various incarnations.  The following list is certainly
incomplete, but serves to recognize the many shoulders on which this
application sits:
.PP
John Bintz,
Arpad Biro,
Daniel Borgmann,
Hans Breuer,
Bulia Byak,
Chema Celorio,
Johan Ceuppens,
Zbigniew Chyla,
Alexander Clausen,
John Cliff,
Kees Cook,
Robert Crosbie,
Jon Cruz,
Danilo Egan,
Frank Felfe,
Fred,
Ted Gould,
Bryce Harrington,
Carl Hetherington,
Nathan Hurst,
Thomas Ingham,
Bob Jamison,
Lauris Kaplinski,
Lynn Kerby,
Petr Kovar,
Raph Levien,
Vitaly Lipatov,
Dmitry G. Mastrukov,
Michael Meeks,
Federico Mena,
MenTaLguY,
Aubanel \s-1MONNIER\s0,
Peter Moulder,
Yukihiro Nakai,
Christian Neumair,
Mitsuru Oka,
Jon Phillips,
Christian Schaller,
Tom von Schwerdtner,
Pat Suwalski,
Adib Taraben,
David Turner,
Daniel Yacob,
Masatake Yamato
.PP
This man page was put together by Bryce Harrington
<brycehar@bryceharrington.com>.
.SH "HISTORY"
.IX Header "HISTORY"
The codebase that would become Inkscape began life in 1999 as the
program Gill, the \s-1GNOME\s0 Illustrator application, created by Raph
Levien.  The stated objective for Gill was to eventually support all of
\&\s-1SVG\s0.  Raph implemented the PostScript bezier imaging model, including
stroking and filling, line cap style, line join style, text, etc.  
Raph's Gill page is at http://www.levien.com/svg/.  Work on Gill appears
to have slowed or ceased in 2000.
.PP
The next incarnation of the codebase was to become the highly popular
program Sodipodi, led by Lauris Kaplinski.  The codebase was turned
into a powerful illustration program over the course of several
year's work, adding several new features, multi-lingual support, porting
to Windows and other operating systems, and eliminating dependencies.
.PP
Inkscape was formed in 2003 by four active Sodipodi developers, Bryce
Harrington, MenTaLguY, Nathan Hurst, and Ted Gould, wanting to take a
different direction with the codebase in terms of focus on
\&\s-1SVG\s0 compliance, interface look-and-feel, and a desire to open
development opportunities to more participants.  
.SH "COPYRIGHT AND LICENSE"
.IX Header "COPYRIGHT AND LICENSE"
\&\fBCopyright (C)\fR 1999\[en]2004 by Authors.  
.PP
\&\fBInkscape\fR is free software; you can redistribute it and/or modify it
under the terms of the \s-1GPL\s0.  


NEWS
Inkscape Release Notes
----------------------
Please see the Inkscape website for the latest news about the project,
or the ChangeLog for further details about releases. This file contains 
the release notes for the current release.


                                 Inkscape 0.40

In brief

   This release has been slow in coming, but it was worth the wait: Inkscape
   0.40 brings a huge amount of new features, usability enhancements,
   documentation, and bugfixes. The three most important new features are:

     * Layers are almost complete in this version; this fills a major gap in
       functionality, making Inkscape usable for complex artwork.

     * The Potrace bitmap tracer is incorporated into Inkscape - now
       vectorizing bitmaps is easy and interactive. A matching command,
       "Create bitmap copy", converts from any object to bitmap.

     * The text-on-path implementation is complete and very flexible. Both
       the text and the path remain fully editable and transformable.

   Many other improvements deserve a brief mention:

     * Features: new document templates, improved freehand and calligraphy
       tools, randomized stars, POV export, some clipboard integration,
       experimental flowText implementation, new command-line options, better
       PNG and EPS export, and more.

     * Usability: new file dialogs, current style for new objects, select
       under & drag selected with Alt, loads of new statusbar tips, new shape
       controls, better default gradients, and more.

     * Documentation: 3 new tutorials (Shapes, Tracing, Calligraphy), many
       translations added and updated.

     * Bugfixes: too many to mention (in total, 228 bugs closed in the
       tracker since 0.39), including several important SVG compatibility
       fixes.

New functionality

     * Support for layers is the most important new feature which is almost
       complete in this version:

          * The active group selector in the statusbar has revealed its true
            identity as the quick layer selector, complete with toggle
            buttons for locking and hiding the current layer. Unfortunately a
            proper layer dialog didn't make it into the release (we've taken
            long enough already), but we're going to rectify this in 0.41 by
            implementing a complete traditional layers dialog.
          * The new Layer menu has commands for creating, renaming, and
            deleting layers, as well as for moving the current layer up or
            down in the z-order of sibling layers (changing the z-order of an
            object moves it only within its parent group or layer, as
            before). Again, more commands will be added in 0.41.
          * Hiding and locking of objects and layers is fully implemented.
            Hiding is done via the display CSS property, so objects hidden in
            Inkscape will be hidden in any compliant SVG renderer. Locking
            prevents an object from being selected using any of the tools. If
            you select an object through other means (for example, with the
            Find dialog or the XML editor), you can change it as usual.
            Hiding or locking groups or layers applies to all objects within
            them.
          * Layers are implemented as SVG groups (the <g> element) with
            inkscape:groupmode="layer". Selection and other commands will
            treat them as layers rather than groups. The possibility
            (introduced in 0.39) to temporarily enter any group as if it were
            a layer remains; such a temporary layer is then shown in the
            layer selector and behaves the same as a normal layer with
            respect to selection commands. Normal (non-temporary) layers can
            also be nested within one another.
          * All the drawing tools as well as Paste refuse to create new
            objects if the current layer is locked or hidden, with an
            appropriate statusbar message.
          * The Ctrl+A (Select All), Tab, and Shift+Tab selection shortcuts
            by default work only in the current layer (not going into either
            parent or children layers) and ignore locked and hidden objects.
            This can be changed in Preferences (the Selecting tab).
          * The Find dialog can optionally limit the search to the current
            layer and include locked and hidden objects in the search (they
            are excluded by default).
          * Selecting an object makes that object's layer current. This
            behavior is somewhat experimental (most other vector editors do
            not do this), but in our testing so far, it is very convenient.
            Just by selecting an object you immediately get into its "local
            context" (i.e. its layer or sublayer) where you can add objects,
            select all within the layer, lock or hide the layer (e.g. to
            reach another layer beneath it), etc.

     * Text on path: Inkscape now provides a complete implementation of SVG
       <textPath> element. Select a text object and a path, shape, or offset
       and do the Text > Put on Path command to make the text follow the
       path. Both text and path remain fully editable (including kerns and
       letterspacing in text). Several texts can be put to one path. When you
       move the path, its attached texts move with it; however, you can move
       the text away from its path or transform it without losing the link.
       The Text > Put on Path command converts a text-on-path into regular
       text.
     * Inkscape can do bitmap tracing (Shift+Alt+B) from the GUI. The only
       tracer included with the program so far is Potrace
       (http://potrace.sf.net) but we may add other tracers in the future.
       Potrace works very well for black-and-white images; in its dialog,
       several tracing options (brightness threshold, edge detection,

debian/inkscape.bash
# inkscape(1) completion 
# put this file in /etc/bash_completion.d/ 
# allali@univ-mlv.fr

have inkscape &&
_inkscape()
{
        local cur

        COMPREPLY=()
        cur=${COMP_WORDS[COMP_CWORD]}

        if [[ "$cur" == -* ]]; then
                COMPREPLY=( $( compgen -W '-? --help -V --version \
			-g --with-gui -p --pipe \
			-o --export-filename= --export-overwrite --export-type=\
			--export-png= -d --export-dpi= -a --export-area= \
			-w --export-width= -h --export-height= -i --export-id= \
			-j --export-id-only  -t --export-use-hints -b --export-background= \
			--action-list --select= --actions= --batch-process --pdf-page= \
			-y --export-background-opacity= ' -- $cur ) )
        else
                _filedir '@(ai|ani|bmp|cur|dia|eps|gif|ggr|ico|jpe|jpeg|jpg|pbm|pcx|pdf|pgm|png|ppm|pnm|ps|ras|sk|svg|svgz|targa|tga|tif|tiff|txt|wbmp|wmf|xbm|xpm)'
        fi

}
[ "${have:-}" ] && complete -F _inkscape $filenames inkscape
